pkg/config/tlscfg/options.go

```diff
@@ -114,7 +114,9 @@ func (p *Options) Config(logger *zap.Logger) (*tls.Config, error) {
 		}
 	}
 
-	go p.certWatcher.watchChangesLoop(tlsCfg.RootCAs, tlsCfg.ClientCAs)
+	p.certWatcher.watchCertPair()
+	p.certWatcher.watchCert(p.certWatcher.opts.CAPath, tlsCfg.RootCAs)
+	p.certWatcher.watchCert(p.certWatcher.opts.ClientCAPath, tlsCfg.ClientCAs)
```

Comment:
what's the reason for calling these three methods externally, instead of doing it all in certWatcher() constructor?

Reply:
There can also be errors in those methods, which are being ignored here. I don't see a good reason for making options.go aware of all these internal details.

Reply:
> what's the reason for calling these three methods externally, instead of doing it all in certWatcher() constructor?

These three methods (and the current `watchChangesLoop`) need `RootCAs` and `ClientCAs` from `tlsCfg`. For now, `tlsCfg` and its fields are initialized in [`func (p *Options) Config`](https://github.com/jaegertracing/jaeger/pull/4260/files/3ff7ee086c435d30f03dcd9f726e21b74fc37b17#diff-0001cdb928045f195b69e9e8546205bdbc96779bb25510f927ae8d6badd3781bL46).

Reply:
I'll make tlsCfg a parameter in certWatcher constructor so that we can call these three all there